### PR TITLE
Add selection API with deselectAll

### PR DESCRIPTION
**Pull request by volfied**
Adding this so we can get rid of performMenuCommand(1016) in design.

### CODE REVIEW

File: src/lib/selection.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ */
+
+define(function (require, exports) {
+    "use strict";
+
+    var PlayObject = require("../playobject");
+
+    /**
+     * Removes the pixel selection
```
Comment (pineapplespatula) [inline src/lib/selection.js]: pixel? does this change not the layer state too?

Comment (volfied) [inline src/lib/selection.js]: This is for marching ants marquee selection, we don't use it in Design, but we get rid of it on startup.

**Comment by mcilroyc** [comment]
I wish we had a better name to distinguish *layer* selection from *marquee* selection.  The name of this module is ambiguous.

**Comment by iwehrman** [comment]
We inadvertently diverged from Photoshop's terminology of "targeted" layers. We could use that instead. I sort of like that.

**Pull request merged**